test/integration/test_path.py

```diff
@@ -101,3 +103,111 @@ def test_write_delete(backend_aware_bucketfs_params, children_poem, classic_poem
     poem_path1.rm()
     expected_names = {"classic", "highlands.txt"}
     assert _collect_all_names(poems_root) == expected_names
+
+
+def test_infer_path_onprem(backend,backend_aware_bucketfs_params):
+    """
+    Creates the PathLike and validates it.
+    """
+    if backend != "saas":
+        pytest.skip("The test runs only with SaaS database")
+    host_port = re.search(
+        r"http://(\d{1,3}(?:\.\d{1,3}){3}):(\d+)", backend_aware_bucketfs_params["url"]
+    )
+    url = infer_path(
+        bucketfs_host=host_port.group(1),
+        bucketfs_port=host_port.group(2),
+        bucketfs_name=backend_aware_bucketfs_params["service_name"],
+        bucket=backend_aware_bucketfs_params["bucket_name"],
+        bucketfs_user=backend_aware_bucketfs_params["username"],
+        bucketfs_password=backend_aware_bucketfs_params["password"],
+        path_in_bucket="onpremtest/",
+        bucketfs_use_https=backend_aware_bucketfs_params["verify"],
+    )
+    assert isinstance(url, bfs.path.BucketPath)
+    assert backend_aware_bucketfs_params["url"] == url._bucket_api._service
+    assert (
+        backend_aware_bucketfs_params["service_name"] == url._bucket_api._service_name
+    )
+    assert backend_aware_bucketfs_params["bucket_name"] == url._bucket_api._name
+    assert "onpremtest" == str(url._path)
+
+
+def test_infer_path_saas(
+    backend, saas_host, saas_pat, saas_account_id, backend_aware_saas_database_id
+):
+    """
+    Creates the SaasBucket with fixture details realted to Saas and validates it.
+    """
+    if backend != "saas":
+        pytest.skip("The test runs only with SaaS database")
+    url = infer_path(
+        saas_url=saas_host,
+        saas_account_id=saas_account_id,
+        saas_database_id=backend_aware_saas_database_id,
+        saas_token=saas_pat,
+        path_in_bucket="saastest/",
+    )
+    assert isinstance(url, bfs.path.BucketPath)
+    assert saas_host == url._bucket_api._url
+    assert saas_account_id == url._bucket_api._account_id
+    assert backend_aware_saas_database_id == url._bucket_api._database_id
+    assert saas_pat == url._bucket_api._pat
+    assert "saastest" in str(url._path)
+
+
+def test_infer_path_and_write(
+    backend,
+    backend_aware_bucketfs_params,
+    children_poem,
+    saas_host,
+    saas_pat,
+    saas_account_id,
+    backend_aware_saas_database_id,
+):
+    print(backend)
+    print(backend_aware_bucketfs_params)
+    print(children_poem)
+    print(saas_host)
+    print(saas_pat)
```

Comment:
Pls do not submit code like this to Github. It has a high risk to leak confidential credentials!